addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -0,0 +1,581 @@
+import org.openhab.binding.somfytahoma.config.SomfyTahomaConfig;
+import org.openhab.binding.somfytahoma.internal.SomfyTahomaException;
+import org.openhab.binding.somfytahoma.internal.discovery.SomfyTahomaItemDiscoveryService;
+import org.openhab.binding.somfytahoma.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.openhab.binding.somfytahoma.SomfyTahomaBindingConstants.*;
+
+/**
+ * The {@link SomfyTahomaBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+public class SomfyTahomaBridgeHandler extends ConfigStatusBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyTahomaBridgeHandler.class);
+
+    private String cookie;
+
+    /**
+     * Future to poll for updated
+     */
+    private ScheduledFuture<?> pollFuture;
+
+    /**
+     * Our configuration
+     */
+    protected SomfyTahomaConfig thingConfig;
+
+    //Gson & parser
+    private final Gson gson = new Gson();
+
+    private SomfyTahomaItemDiscoveryService discoveryService = null;
+
+    public SomfyTahomaBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        String thingUid = getThing().getUID().toString();
+        thingConfig = getConfigAs(SomfyTahomaConfig.class);
+        thingConfig.setThingUid(thingUid);
+
+        login();
+        scheduler.schedule(() -> startDiscovery(), 1, TimeUnit.SECONDS);
+        initPolling(thingConfig.getRefresh());
+    }
+
+    /**
+     * starts this things polling future
+     */
+    private void initPolling(int refresh) {
+        stopPolling();
+        pollFuture = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    updateTahomaStates();
+                } catch (Exception e) {
+                    logger.debug("Exception during poll!", e);
+                }
+            }
+        }, 10, refresh, TimeUnit.SECONDS);
+
+    }
+
+    private synchronized void login() {
+        String url = null;
+
+        if (StringUtils.isEmpty(thingConfig.getEmail()) || StringUtils.isEmpty(thingConfig.getPassword())) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device as username and/or password are null");
+            return;
+        }
+
+        try {
+            url = TAHOMA_URL + "login";
```

Comment:
Unfortunately I can't. I need access also to header fields for getting cookie and header is available via HttpsUrlConnection object.